### ISSUE TITLE
Can't install ghost npm package w/ recent versions of npm@3.

### ISSUE THREAD

**Issue by jaswilli** [TryGhost/Ghost#6691]
### Issue Summary

Unable to install the ghost npm package with npm version 3. It appears as though the `preinstall` script has run afoul of the way that npm 3 behaves. After a cursory look at what's going on, it appears as though it gets into an endless recursive install and just keeps on trying to `npm install semver` over and over.

### Steps to Reproduce

1. In an otherwise unused directory, do an `npm install ghost`.

### Technical details:

* Ghost Version: 0.7.8 (npm package)
* Node Version: 4.4.x
* Browser/OS: OS X; Linux
* Database: n/a

**Comment by ErisDS** [comment]
Yeah, this is a real problem and I don't know what to do about it. 

It is a bug in npm: https://github.com/npm/npm/issues/10379#issuecomment-163316532 and they intend to change the behaviour back to what it should be. So essentially at the moment, there are certain "broken" versions of npm for which Ghost cannot be installed from npm.

If you have any thoughts on how we could workaround it, I'd be interested. However, I tend to be of the attitude that if other people break their stuff we should not work around it - else there's no incentive to fix it or not break things in future.

**Comment by jaswilli** [comment]
Interesting. I've been assuming it was the `npm install semver` part of the `preinstall` script that was causing trouble and we could potentially work around it by using `bundledDependencies`, but after reading that issue you linked it seems like maybe not (and I just published a test version to npm and it still fails--just in a different way).

So my only other non-crazy idea would be to drop the `semver` dependency and do our own version checking. Checking against `"node": "~0.10.0 || ~0.12.0 || ^4.2.0"` shouldn't really be that difficult. I just checked and `preinstall` does work for calling into `startup-check.js`, we just can't do an `npm install` or depend on `semver`.

> I tend to be of the attitude that if other people break their stuff we should not work around it - else there's no incentive to fix it or not break things in future.

You know I feel the same way 😁  -- except maybe when it's npm because that doesn't seem like a standoff I can win.
